rasa/core/processor.py

```diff
@@ -297,7 +297,9 @@ def _get_action(self, action_name):
     ) -> None:
 
         if message.parse_data:
+            parse_message = await self._parse_message(message)
             parse_data = message.parse_data
+            parse_data.update(parse_message)
```

Comment:
why this change? how is it related to the changes to `botframework.py`?

Reply:
Because when I receive an attachement, I set `parse_data` with it in the UserMessage ([line 216](https://github.com/RasaHQ/rasa/pull/4135/files#diff-d27075473e03e27edac04ffe9aa72fb6R216)). 

However, the processor doesn't parse the message if `parse_data` contains a value, leading to an error [when accessing the keys](https://github.com/RasaHQ/rasa/blob/master/rasa/core/processor.py#L309) `intent` and `entities`. I chose to execute `_parse_message` even if the message contains an attachement because some text can be sent with the attachement and it's interesting to be able to parse it alongside the uploaded file.

Reply:
```suggestion
    attachment = postdata['attachments'][0]
    user_msg = UserMessage(
    text=postdata['text'] if postdata.get('text') else "",
    parse_data={
        "attachment": attachment,
        "intent": {"confidence": 1.0, "name": "attachment"},
        "entities": [],
        "text": postdata['text'] if postdata.get('text') else "",
    },
    output_channel=out_channel,
    sender_id=postdata["from"]["id"],
    input_channel=self.name(),
)
```
Otherwise we could do something like this. What do you think about it?

Reply:
but the point of this check is to _not_ parse a message again that already contains `parse_data()`

Reply:
Yup, fixed